org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/wizard/project/ProjectPreviewWizardPage.java

```diff
@@ -376,14 +391,21 @@ private void populateTree(final OmniGradleBuildStructure buildStructure) {
 
                 @Override
                 public void run() {
-                    ProjectPreviewWizardPage.this.projectPreviewTree.removeAll();
-
-                    // populate the tree from the build structure
-                    OmniGradleProjectStructure rootProject = buildStructure.getRootProject();
-                    TreeItem rootTreeItem = new TreeItem(ProjectPreviewWizardPage.this.projectPreviewTree, SWT.NONE);
-                    rootTreeItem.setExpanded(true);
-                    rootTreeItem.setText(rootProject.getName());
-                    populateRecursively(rootProject, rootTreeItem);
+                    try {
+                        if (!ProjectPreviewWizardPage.this.projectPreviewTree.isDisposed()) {
+                              ProjectPreviewWizardPage.this.projectPreviewTree.removeAll();
+
+                              // populate the tree from the build structure
+                              OmniGradleProjectStructure rootProject = buildStructure.getRootProject();
+                              TreeItem rootTreeItem = new TreeItem(ProjectPreviewWizardPage.this.projectPreviewTree, SWT.NONE);
+                              rootTreeItem.setExpanded(true);
+                              rootTreeItem.setText(rootProject.getName());
+                              populateRecursively(rootProject, rootTreeItem);
+                        }
+                    } catch (SWTException e) {
```

Comment:
I don't think you should catch SWTException here. Since you check for the disposed widget before. Anything goes wrong after that, I'd rather get it popped up, so we notice it and fix it. Simply logging the error, we might lose track of potential errors in the future.

btw it doesn't necessarily means `Failed to properly close ProjectPreviewWizardPage` when that happens, even if there's a good chance it does.